### PR TITLE
Document next/link component and create stories

### PR DESCRIPTION
**Pull request by valentinpalkovic**
## What I did

I have created Stories to cover several functionalities of `next/link`.

## How to test

- [x] Is this testable with Jest or Chromatic screenshots?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: code/frameworks/nextjs/src/config/webpack.ts [hunk 1]
```diff
@@ -42,6 +42,9 @@ const setupRuntimeConfig = (baseConfig: WebpackConfig, nextConfig: NextConfig):
    * In Next13, the `newNextLinkBehavior` option now defaults to truthy (still
    * `undefined` in the config), and `next/link` was engineered to opt *out*
    * of it
+   *
+   * TODO: Revisit logic as soon as the changes in https://github.com/vercel/next.js/pull/42623
```
Comment (kylegach) [inline code/frameworks/nextjs/src/config/webpack.ts:46]: I tried to write the logic in a way that would work before and after that change is released. But probably a good idea to revisit, regardless.

File: code/frameworks/nextjs/template/stories_default-js/Link.stories.jsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      </Link>
+    </li>
+  </ul>
+);
+
+export default {
+  component: Component,
+};
+
+export const Default = {};
```
Comment (kylegach) [inline code/frameworks/nextjs/template/stories_default-js/Link.stories.jsx:74]: This is a great addition! Could we add a play function to test that the various props actually work?

Comment (kylegach) [reply code/frameworks/nextjs/template/stories_default-js/Link.stories.jsx:74]: Or, perhaps just passing `style` would do it, assuming these are tested in Chromatic?

Comment (valentinpalkovic) [reply code/frameworks/nextjs/template/stories_default-js/Link.stories.jsx:74]: Can I assert the `action` directly?

Something like:

```js
expect(action).toHaveBeenCalledWith(...)
```

Comment (kylegach) [reply code/frameworks/nextjs/template/stories_default-js/Link.stories.jsx:74]: Yep! It's more like `expect(context.args.onClick).toHaveBeenCalledWith(...)`, though. All args with actions are auto-spy'd.

Comment (valentinpalkovic) [reply code/frameworks/nextjs/template/stories_default-js/Link.stories.jsx:74]: But in this case, the action is not passed as an argument. Instead, the decorator uses `@storybook/action-addons` internally. It is not injected per dependency injection. So my question is, whether 

```
import type { action } from '@storybook/addon-actions';
```

is mocked in general in all play functions.

@yannbf, maybe you know more.

**Comment by ndelangen** [comment]
@valentinpalkovic can we merge this?

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
Looks like nothing remained after resolving merge conflicts..

I'll close it, please re-open if there's still something here..

**Pull request closed without merging**